Assets/Scripts/EnemyController.cs

```diff
@@ -0,0 +1,193 @@
+        }
+
+        wanderArea.FreezeArea();
+        _timer = wanderTimer;
+        // remember initial speed
+        _initialSpeed = _agent.speed;
+    }
+
+    /// <summary>
+    /// Moves around randomly in the defined wander area at a defined frequency and checks for a 
+    /// target near. If a target is near, starts chasing (possibly attacking) it until the aggro is lost.
+    /// </summary>
+    private void Update()
+    {
+        if (!_isChasing) //readability
+            Wander();
+
+        if (isAggressive)
+        {
+            // Check for a target near
+            float distanceToTarget = Vector3.Distance(_target.position, transform.position);
+            if (distanceToTarget <= lookRadius || _isChasing)
+            {
+                Chase(distanceToTarget);
+            }
+
+            // stop chasing
+            if (_isChasing && distanceToTarget >= escapeRadius)
+            {
+                _agent.speed = _initialSpeed;
+                _isChasing = false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Wanders in wander area.
+    /// </summary>
+    private void Wander()
+    {
+        _timer += Time.deltaTime;
+        if (_timer >= wanderTimer)
+        {
+            MoveToNextPoint();
+            _timer = 0;
+        }
+    }
+
+    /// <summary>
+    /// Chases the target.
+    /// </summary>
+    /// <param name="distanceToTarget">The distance between NPC and its target</param>
+    private void Chase(float distanceToTarget)
+    {
+        _isChasing = true;
+        _agent.speed = runningSpeed;
+        _agent.SetDestination(_target.position);
+
+        if (distanceToTarget <= _agent.stoppingDistance)
+        {
+            FaceTarget();
+            // attack
+        }
+    }
+
+    /// <summary>
+    /// Starts moving to a new random point in the wander area or goes back to wander area if outside.
+    /// </summary>
+    private void MoveToNextPoint()
+    {
+        // move back to wander area after chasing, also in this case:
+        /* at very high speed and acceleration the NPC can get out of the wander area because 
+         * the NavMeshAgent behaves in a physical manner (inertia) */
+        if (!_wanderAI.IsInArea(wanderArea, transform.position))
+        {
+            _agent.SetDestination(wanderArea.GetCenterPosition());
+        }
+        else
+        {
+            NavMeshPath wanderPath = _wanderAI.GetNextWanderPath(wanderArea, _agent, moveLayer.value);
+            _agent.SetPath(wanderPath);
+        }
+    }
+
+    /// <summary>
+    /// Rotates to face a target.
+    /// </summary>
+    private void FaceTarget()
+    {
+        Vector3 direction = (_target.position - transform.position).normalized;
+        Quaternion lookRotation = Quaternion.LookRotation(new Vector3(direction.x, 0f, direction.z));
+        transform.rotation = Quaternion.Slerp(transform.rotation, lookRotation, Time.deltaTime * rotationSpeed);
```

Comment:
FYI: This slerp will never finish due to dividing the difference every time. Shouldn't be a problem or slow but checking if the rotation difference is smaller than something (like 0.01 or so) before executing it, could save some calculation